src/Event/AbstractStorageEventListener.php

```diff
@@ -137,16 +139,9 @@ public function buildFilename($table, $entity) {
  * @return string
  */
 	public function buildPath($table, $entity) {
-		$path = '';
-		if ($this->_config['tableFolder']) {
-			$path .= $table->table() . DS;
-		}
-		if ($this->_config['randomPath'] === true) {
-			$path .= StorageUtils::randomPath($entity[$table->primaryKey()]);
-		}
-		if ($this->_config['uuidFolder'] === true) {
-			$path .= $this->stripDashes($entity[$table->primaryKey()]) . DS;
-		}
+		$pathBuilder = $this->createPathBuilder($entity['adapter']);
```

Comment:
@burzum 

>because the "adapter" (which should have been named config in the past) is a configuration name that can be anything. For example you could have "Local1" and "Local2" to store files into different base path locations (/var/files1 & /var/files2/ for example). It was never thought to be used as a convention.

I'm afraid I can't agree with that. `adapter` property **has** to be a valid storage adapter name. New listeners depend on that. See [this](https://github.com/burzum/cakephp-file-storage/blob/3.0-event-overhaul/src/Storage/Listener/LocalListener.php#L82) and [this](https://github.com/burzum/cakephp-file-storage/blob/3.0-event-overhaul/src/Storage/Listener/ImageProcessingTrait.php#L121)